i2c/ssd1306_i2c/ssd1306_i2c.c

```diff
@@ -387,13 +387,23 @@ int main() {
         "by the name of",
         "    PICO"
     };
-
     int y = 0;
-    for (uint i = 0 ;i < count_of(text); i++) {
+    for (uint i = 0; i < count_of(text); i++) {
         WriteString(buf, 5, y, text[i]);
-        y+=8;
+        y += 8;
+        // Height limit reached. Show some lines.
+        if (y == SSD1306_HEIGHT) {
+            render(buf, &frame_area);
+            sleep_ms(3000);
+            memset(buf, 0, SSD1306_BUF_LEN);
+            y = 0;
+        }
+    }
+    // Check if there's any more text left to display.
+    if (y != 0) {
+        render(buf, &frame_area);
+        sleep_ms(3000);
```

Comment:
Probably don't need the `sleep_ms(3000);` here, because there's already a `sleep_ms(3000);` on line 409 ?

Reply:
I think we should leave this one and remove the one on line 409

Let me leave another comment on line 399 to explain